internal/utils.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package internal
+
+import "sync"
+
+// LockMap uses an RWMutex to synchronize map access to allow for concurrent access.
+// This should not be used for cases with heavy write load and performance concerns.
+type LockMap struct {
+	sync.RWMutex
+	m map[string]string
+}
+
+func NewLockMap(m map[string]string) *LockMap {
+	return &LockMap{m: m}
+}
+
+func (l *LockMap) Iter(f func(key string, val string)) {
```

Comment:
Might want to note that `Iter` is read only. It should be obvious, but good to be explicit.